test-common/utils.c

```diff
@@ -64,13 +64,21 @@ eos_updater_fixture_setup (EosUpdaterFixture *fixture,
 
   g_test_message ("Using fixture directory ‘%s’", tmpdir_path);
 
-  source_gpg_home_path = g_test_build_filename (G_TEST_DIST, "gpghome", NULL);
+  source_gpg_home_path = g_build_filename (top_srcdir, "tests", "gpghome", NULL);
   fixture->gpg_home = create_gpg_keys_directory (fixture->tmpdir, source_gpg_home_path);
 }
 
+void
+eos_updater_fixture_setup (EosUpdaterFixture *fixture,
+                           gconstpointer user_data G_GNUC_UNUSED)
+{
+  g_autofree gchar *top_srcdir = g_test_build_filename (G_TEST_DIST, "..", NULL);
```

Comment:
True, it is a bit fragile, but I think a full fix would involve reworking `eos_test_add()` and being a bit more explicit about the separation between different test fixtures. I think that’s a bit out of scope for this branch. I’m happy to leave this as fragile for now, and refactor it later if it causes a problem in future (it should fairly obviously cause problems if `eos_test_add` is used from the wrong directory, since the GPG files for the test fixture won’t be found).